BAT/BCEngineMCMC.h

```diff
@@ -1411,13 +1450,29 @@ class BCEngineMCMC
     /** @} */
 
 private:
+
+    // grant friendship to BCModel to access the ThreadLocalStorage
+    friend class BCModel;
+    
     /**
      * Keeps variables that need to be both members and thread local.
      */
     struct ThreadLocalStorage {
         /**
          * Store local proposal point  */
-        std::vector<double> xLocal;
+        std::vector<double> parameters;
+
+        /**
+         * Store log_prior of local proposal point */
+        double log_prior;
```

Comment:
I don't like this coupling with `BCModel` and the prior. We worked hard to separate things a bit and keep the prior out of the engine